tree/ntuple/v7/inc/ROOT/RNTupleMerger.hxx

```diff
@@ -34,10 +46,67 @@ class RFieldMerger {
 private:
    /// The merged field descriptor
    RFieldDescriptor fMergedField = RFieldDescriptor();
+
 public:
    static RResult<RFieldMerger> Merge(const RFieldDescriptor &lhs, const RFieldDescriptor &rhs);
 };
 
+// clang-format off
+/**
+ * \class ROOT::Experimental::RNTupleMerger
+ * \ingroup NTuple
+ * \brief Given a set of RPageSources merge them into an RPageSink
+ */
+// clang-format on
+class RNTupleMerger {
+
+public:
+   /// Merge a given set of sources into the destination
+   void Merge(std::vector<std::unique_ptr<RPageSource>> &sources, std::unique_ptr<RPageSink> &destination);
+
+private:
+   // Struct to hold column and field descriptors
+   struct RColumnInfo {
```

Comment:
It seems that we don't really need the full field and column descriptor. Wouldn't it be sufficient to store a pair of source column ID and destination column ID?

Reply:
I simplified `RColumnInfo` accordingly.